src/main/java/com/google/sps/data/Chat.java

```diff
@@ -6,34 +6,32 @@
 import com.google.cloud.datastore.FullEntity;
 import com.google.cloud.datastore.Query;
 import com.google.cloud.datastore.QueryResults;
+import com.google.cloud.datastore.StructuredQuery.CompositeFilter;
+import com.google.cloud.datastore.StructuredQuery.PropertyFilter;
 import com.google.cloud.datastore.StructuredQuery.OrderBy;
 import com.google.cloud.datastore.Key;
 import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.datastore.IncompleteKey;
+import com.google.cloud.datastore.Value;
+import com.google.cloud.datastore.StringValue;
 import com.google.gson.Gson;
 import com.google.sps.data.Chat;
 
 import java.io.IOException;
 import java.util.*;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.stream.Collectors;
-import javax.servlet.annotation.WebServlet;
-import javax.servlet.http.HttpServlet;
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
-import org.jsoup.Jsoup;
-import org.jsoup.safety.Whitelist;
 import java.lang.String;
 
 /** An item on a todo list. */
 public final class Chat {
 
   private long id;
-  private List<String> participants;
-  private List<String> messages;
+  private List<StringValue> participants;
+  private List<StringValue> messages;
```

Comment:
Keep these as a `List<String>` and instead use streams to transform the results from `entity.getList` to a List of String.

```suggestion
  private List<String> participants;
  private List<String> messages;
```

Reply:
This is tricky. I've gotten it to a point where it works, or is close to it, you sure I need to do that?

Reply:
Actually I'll try it out in a new commit